www/js/main.js

```diff
@@ -4625,69 +4625,82 @@ var showHome = ( function() {
                 name = button.siblings( "[id='station_" + id + "']" ),
                 showSpecialOptions = function( value ) {
 					var opts = select.find( "#specialOpts" ),
-						hex = controller.special && controller.special.hasOwnProperty( id ) ? controller.special[ id ].sd : "",
-
-						// Remember station type to understand special data format
-						hexFormat  = controller.special && controller.special.hasOwnProperty( id ) ? controller.special[ id ].st : 0;
+						data = controller.special && controller.special.hasOwnProperty( id ) ? controller.special[ id ].sd : "",
+						type  = controller.special && controller.special.hasOwnProperty( id ) ? controller.special[ id ].st : 0;
 
 					opts.empty();
 
-					if ( value === 1 ) {
+					if ( value === 0 ) {
+						opts.append(
+							"<p class='special-desc center small'>" +
+								_( "Select the station type using the dropdown selector above and configure the station properties." ) +
+							"</p>"
+						).enhanceWithin();
+					} else if ( value === 1 ) {
+						data = ( type === value ) ? data : "0000000000000000";
+
 						opts.append(
 							"<div class='ui-bar-a ui-bar'>" + _( "RF Code" ) + ":</div>" +
-							"<input class='center' data-corners='false' data-wrapper-class='tight ui-btn stn-name' id='rf-code' type='text' value='" + hex + "'>"
+							"<input class='center' data-corners='false' data-wrapper-class='tight ui-btn stn-name' id='rf-code' required='true' type='text' value='" + data + "'>"
 						).enhanceWithin();
 					} else if ( value === 2 ) {
-						var data = parseRemoteStationData( hex );
+						data = ( type === value ) ? parseRemoteStationData( data ) : parseRemoteStationData( "00000000005000" );
```

Comment:
Sorry if it's being nit picky but do you think this is easier to read?

`data = parseRemoteStationData( type === value ? data : "00000000005000" );`